### PR TITLE
feat(BBD-648): Add query

### PR DESCRIPTION
**Pull request by Vic-Dev**

### CODE REVIEW

File: src/record-count/index.ts [hunk 1]
```diff
@@ -12,13 +12,17 @@ class RecordCount {
   };
 
   init() {
+    this.state.query = Selectors.currentQuery(this.flux.store.getState());
```
Comment (effervescentia) [inline src/record-count/index.ts]: We should put this in a member variable declaration, like:

```js
class RecordCount {

  state: RecordCount.State = {
    query: Selectors.currentQuery(this.flux.store.getState())
  };
}
```

File: test/unit/record-count.ts [hunk 2]
```diff
@@ -52,4 +68,18 @@ suite('RecordCount', ({ expect, spy, itShouldBeConfigurable, itShouldHaveAlias }
       expect(set).to.be.calledWith({ to, from, hasResults: true });
     });
   });
+  describe('updateQuery()', () => {
```
Comment (effervescentia) [inline test/unit/record-count.ts]: add a newline above here

File: test/unit/record-count.ts [hunk 3]
```diff
@@ -19,10 +19,26 @@ suite('RecordCount', ({ expect, spy, itShouldBeConfigurable, itShouldHaveAlias }
   });
 
   describe('init()', () => {
+    it('should update query', () => {
+      const query = 'eyyo';
+      const getState = spy();
+      stub(Selectors, 'currentQuery').returns(query);
+      recordCount.flux = <any>{ on: stub(), store: { getState } };
+      recordCount.expose = () => null;
```
Comment (jmbtutor) [inline test/unit/record-count.ts]: I don't think this is needed anymore. Same with line 40.

File: test/unit/record-count.ts [hunk 4]
```diff
@@ -19,10 +19,26 @@ suite('RecordCount', ({ expect, spy, itShouldBeConfigurable, itShouldHaveAlias }
...
+
+      expect(getState).to.be.called;
+      expect(recordCount.state.query).to.eq(query);
+    });
+
     it('should listen for events', () => {
       const on = spy();
-      recordCount.flux = <any>{ on };
+      stub(Selectors, 'currentQuery');
+      recordCount.flux = <any>{ on, store: { getState: stub() } };
```
Comment (jmbtutor) [inline test/unit/record-count.ts]: `() => null` instead of `stub()`

**Pull request merged**